### PR TITLE
feat: Initial version of the SDK.

### PR DESCRIPTION
**Pull request by richieshilton**
### Summary 

This PR is the first version of the SDK. 

### Tasks

- Add data capture.
- Add public methods.
- Add Event gathering and sending.
- Add public header. 
- Update readme

### CODE REVIEW

File: Sources/AirwallexRisk/Extensions/UserDefaults+Extensions.swift [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+//  UserDefaults+Extensions.swift
+//  
+//
+//  Created by Richie Shilton on 7/7/2023.
+//
+
+import Foundation
+
+extension UserDefaults {
+    static let sdk = UserDefaults(suiteName: "airwallex-risk-sdk")
```
Comment (JinW-Airwallex) [inline Sources/AirwallexRisk/Extensions/UserDefaults+Extensions.swift:11]: might need to prefix suiteName with some sort of `com.airwallex.xxx`?

File: Sources/AirwallexRisk/AirwallexRisk.swift [hunk 2]
```diff
@@ -1,6 +1,54 @@
...
+//
 
-    public init() {
+import Foundation
+
+public class AirwallexRisk {
+    static var shared: AirwallexRisk = .init()
+    private(set) var context: AirwallexRiskContext?
+
+    /// Must be called once as early as possible in the app lifecycle.
```
Comment (JinW-Airwallex) [inline Sources/AirwallexRisk/AirwallexRisk.swift:15]: I thought the doc syntax should be something like 
```
/**
*/
```

File: Sources/AirwallexRisk/Helpers/Storage.swift [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    private let defaultValue: T
+    private let defaults: UserDefaults?
+
+    /// Property wrapper to store Codable objects in UserDefaults.
+    /// If no value is set, `defaultValue` will be stored to UserDefaults for subsequent usage.
+    ///
+    init(
+        key: String,
+        defaultValue: T,
+        defaults: UserDefaults? = .sdk
```
Comment (JinW-Airwallex) [inline Sources/AirwallexRisk/Helpers/Storage.swift:24]: Would it make more sense to be non-optional?

File: Sources/AirwallexRisk/AirwallexRiskContext.swift [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+//  AirwallexRisk
+//
+//  Created by Richie Shilton on 5/7/2023.
+//  Copyright © 2023 Airwallex. All rights reserved.
+//
+
+import Foundation
+
+class AirwallexRiskContext {
+    @Storage(key: AirwallexUserDefaultKey.user, defaultValue: .init())
```
Comment (JinW-Airwallex) [inline Sources/AirwallexRisk/AirwallexRiskContext.swift:12]: As much as i like property wrapper, it makes this class impossible to test

File: Sources/AirwallexRisk/AirwallexRisk.swift (hunk 2 above)
Comment (richieshilton) [reply Sources/AirwallexRisk/AirwallexRisk.swift:15]: yeah, I will need to clean up all the docs

File: Sources/AirwallexRisk/Helpers/Storage.swift (hunk 3 above)
Comment (richieshilton) [reply Sources/AirwallexRisk/Helpers/Storage.swift:24]: The `.init(suiteName:)` is unfortunately optional

File: Sources/AirwallexRisk/AirwallexRiskContext.swift (hunk 4 above)
Comment (richieshilton) [reply Sources/AirwallexRisk/AirwallexRiskContext.swift:12]: Will leave, as discussed offline. Tested the integration.

Comment (richieshilton) [reply Sources/AirwallexRisk/AirwallexRiskContext.swift:14]: I've removed the decoration and will init the Storage objects in the Context init - so we can pass in the UserDefault suite explicitly.

File: Sources/AirwallexRisk/Extensions/UserDefaults+Extensions.swift [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+//  Created by Richie Shilton on 7/7/2023.
+//
+
+import Foundation
+
+extension UserDefaults {
+    private static let airwallexRiskSuiteName = "com.airwallex.risk-sdk"
+
+    /// AirwallexRisk UserDefault instance.
+    static var sdk = UserDefaults(suiteName: airwallexRiskSuiteName)
```
Comment (JinW-Airwallex) [inline Sources/AirwallexRisk/Extensions/UserDefaults+Extensions.swift:14]: I think `let` works here

**Review by JinW-Airwallex — APPROVED** [review]

**Comment by JinW-Airwallex** [comment]
Absolutely well implemented. Nice job!

**Pull request merged**